src/components/UnitContainer.js

```diff
@@ -19,12 +19,18 @@ export default class UnitContainer extends Component {
 
   update() {
     this.setState({
-      units: engine.boardState.getUnits().map(unit => <Unit progress={unit.progress} key={generateUUID()} />)
+      units: engine.boardState.getUnits().map(unit =>
+        <Unit
+          progress={unit.movementPattern.currentLocation}
+          color={unit.player.color}
+          key={generateUUID()}
+          />
+      )
     })
   }
 
   addUnit() {
-    engine.boardState.addUnit()
+    engine.spawnFriendlyUnit()
```

Comment:
True... But the best alternative I can think of is giving Player some spawnUnit() method where it creates the unit with a UnitFactory (cause Player shouldn't know about movement patterns and such) and adds it to board.